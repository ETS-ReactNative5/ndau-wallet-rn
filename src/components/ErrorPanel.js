import React, { Component } from 'react'

import { StyleSheet, Text, View } from 'react-native'

class ErrorPanel extends Component {
  render () {
    return (
      <View style={styles.errorOuterContainer}>
        <View style={styles.errorContainer}>
          <Text style={styles.errorText}>{this.props.errorText}</Text>
        </View>
      </View>
    )
  }
}

var styles = StyleSheet.create({
  errorText: {
    color: '#f75f4b',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular',
    padding: '4%'
  },
  errorContainer: {
    backgroundColor: '#f5d8d1',
    borderWidth: 2,
    borderColor: '#f75f4b',
    borderRadius: 3
  },
  errorOuterContainer: {
    backgroundColor: '#f5d8d1',
    borderWidth: 1,
    borderColor: '#f5d8d1',
    borderRadius: 3
  }
})

export default ErrorPanel
