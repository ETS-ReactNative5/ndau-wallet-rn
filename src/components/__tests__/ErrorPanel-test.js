import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import ErrorPanel from '../ErrorPanel';

import renderer from 'react-test-renderer';

describe('testing ErrorPanel...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<ErrorPanel parentStyles={styles} navigator={navigator} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
