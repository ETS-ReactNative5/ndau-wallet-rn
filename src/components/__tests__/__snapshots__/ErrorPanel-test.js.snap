// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ErrorPanel... renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f5d8d1",
      "borderColor": "#f5d8d1",
      "borderRadius": 6,
      "borderWidth": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f5d8d1",
        "borderColor": "#f75f4b",
        "borderRadius": 6,
        "borderWidth": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#f75f4b",
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 20,
          "padding": "4%",
        }
      }
    />
  </View>
</View>
`;
